core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -473,6 +474,12 @@ object SparkSubmit extends CommandLineUtils {
       OptionAssigner(args.driverExtraLibraryPath, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
         sysProp = "spark.driver.extraLibraryPath"),
 
+      // Standalone only - propagate attributes for dependency resolution at the driver side
+      OptionAssigner(args.packages, STANDALONE, CLUSTER, sysProp = "spark.jars.packages"),
```

Comment:
You can merge this with mesos cluster `OptionAssigner` like `OptionAssigner(args.packages, STANDALONE | MESOS, CLUSTER, sysProp = "spark.jars.packages")`.